tgext/utils/storage.py

```diff
@@ -0,0 +1,32 @@
+import random
+import string
+from os import path, mkdir
+
+from tg import config
+
+
+def store(data):
+    extension = tell_extension(data)
+    filename = generate_id(extension)
+    destination = _specify_path(filename)
```

Comment:
I suggest we use the `id` as a directory and preserve filenames after passing them through http://turbogears.readthedocs.io/en/latest/reference/classes.html#tg.util.files.safe_filename

So in case the user uploads a file name `tg.png` it gets saved as `storage/32132131/tg.png` and preserves the `tg.png` name when downloaded.